pkg/controllers/allocation/scheduling/scheduler.go

```diff
@@ -102,13 +102,15 @@ func globalRequirements(instanceTypes []cloudprovider.InstanceType) (requirement
 		v1.LabelInstanceTypeStable: sets.NewString(),
 		v1.LabelTopologyZone:       sets.NewString(),
 		v1.LabelArchStable:         sets.NewString(),
-		v1.LabelOSStable:           sets.NewString(),
 	}
 	for _, instanceType := range instanceTypes {
+		scheduledZoneSet := sets.String{}
+		for _, offering := range instanceType.Offerings() {
+			scheduledZoneSet.Insert(offering.Zone)
+		}
 		supported[v1.LabelInstanceTypeStable].Insert(instanceType.Name())
-		supported[v1.LabelTopologyZone].Insert(instanceType.Zones().UnsortedList()...)
+		supported[v1.LabelTopologyZone].Insert(scheduledZoneSet.UnsortedList()...)
 		supported[v1.LabelArchStable].Insert(instanceType.Architecture())
-		supported[v1.LabelOSStable].Insert(instanceType.OperatingSystems().UnsortedList()...)
```

Comment:
We might still need this if other cloud providers don't support Windows on all instance types.

Reply:
We can add it back then. I don't even know what it means for an instance type to not support a certain OS, that would be such an artificial restriction for the cloud provider.

Reply:
(then meaning at that time)